cmd/protoc-gen-gorums/dev/strictordering.go

```diff
@@ -0,0 +1,165 @@
+package dev
+
+import (
+	"context"
+	"math"
+	"math/rand"
+	"sync"
+	atomic "sync/atomic"
+	"time"
+
+	"github.com/relab/gorums"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/codes"
+	status "google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type strictOrderingResult struct {
+	nid   uint32
+	reply *anypb.Any
+	err   error
+}
+
+type requestHandler func(*gorums.Message) *gorums.Message
+
+type strictOrderingManager struct {
+	msgID    uint64
+	recvQ    map[uint64]chan *strictOrderingResult
+	recvQMut sync.RWMutex
+}
+
+func newStrictOrderingManager() *strictOrderingManager {
+	return &strictOrderingManager{
+		recvQ: make(map[uint64]chan *strictOrderingResult),
+	}
+}
+
+func (m *strictOrderingManager) createStream(node *Node, backoff backoff.Config) *strictOrderingStream {
```

Comment:
Would it not be better to have this as a method on `Node` and pass `m *receiveQueue` as an argument.

Reply:
Something like this:
```go
func (node *Node) createStream(m *strictOrderingManager, backoff backoff.Config) {
	node.strictOrdering = &strictOrderingStream{
		node:      node,
		nextMsgID: m.nextMsgID,
		backoff:   backoff,
		rand:      rand.New(rand.NewSource(time.Now().UnixNano())),
		sendQ:     make(chan *gorums.Message),
		recvQ:     m.recvQ,
		recvQMut:  &m.recvQMut,
	}
}
```